### ISSUE TITLE
Display error in options

### ISSUE THREAD

**Issue by Greenturtle116** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#250]
I just downloaded the MOD.  When setting things up in the options menu I have a problem. Under Policies & Restrictions under the sub-section 'At Junctions' all of the options are just one jumbled line of text as if each of the many options are typed on top of each other so you can't read what any of the text says or select individual options. Same problem under the Maintenance tab, 'Activated Features' subsection. It also looks like all 8 options are just written on top of each other so I can't toggle them on or off individually.  Anyone else seen this problem before?

**Comment by Greenturtle116** [comment]
![AdvancedFeatures](https://user-images.githubusercontent.com/48950911/55022115-4d4a0800-4fd0-11e9-9c8f-f3c30b6af85c.JPG)
![AtJunctions](https://user-images.githubusercontent.com/48950911/55022116-4de29e80-4fd0-11e9-8fa8-c4b4aa5a20fe.JPG)

Trying to attach screen shots of the two problems I'm having with the options.  You can also see the other MODS I have downloaded in case someone sees an incompatibility issue I've missed.

**Comment by aubergine10** [comment]
This looks like a new issue. I know the code rendering those options was changed in this release to allow for indented options.

Was it like this when first opening the options screen, on only on subsequent openings?

**Comment by VictorPhilipp** [comment]
This issue only occurs when the options dialog is opened from the main menu.

**Comment by aubergine10** [comment]
As the options are only valid in-game, is there some way we can show different options screen if opened from main menu?

It should be possible to determine if it's from main menu based on the stuff present on UIView?

**Comment by 1Tomber** [comment]
When someone is looking into this bug: I loose the display of all settings when changing the language in the TM:PE options menu in-game. Restarting brings it back. Changing it in the main menu is fine.

**Comment by Greenturtle116** [comment]
aubergine10 - this issue presented itself the first time I opened the menus and has continued on subsequent openings of the menu's.  I even did a computer restart for good measure.

**Comment by aubergine10** [comment]
TPB posted some code showing alternate ways to indent checkboxes (which would avoid turning the auto-layout feature off on the parent container)....

# Approach 1: Use a group, and indent the group

```csharp
public void OnSettingsUI(UIHelperBase helper) {
    UIHelperBase group = helper.AddGroup("Mod");
    UIPanel panel = (UIPanel)((UIPanel)((UIHelper)group).self).parent;
    panel.Find<UIPanel>("Content").autoLayoutPadding = new RectOffset(100, 0, 0, 0);
    UICheckBox checkbox = group.AddCheckbox("MyCheckbox", true, (b) => { }) as UICheckBox;
}
```

![v1](https://user-images.githubusercontent.com/1386719/55296528-b7b8d900-5412-11e9-99b0-dd81b36bebcc.png)

# Approach 2: Set relative position

```csharp
public void OnSettingsUI(UIHelperBase helper) {
    UIHelperBase group = helper.AddGroup("Mod");
    UICheckBox checkbox = group.AddCheckbox("My Checkbox", true, (b) => { }) as UICheckBox;
    checkbox.Find<UISprite>("Unchecked").relativePosition += new Vector3(100.0f, 0.0f);
    checkbox.Find<UILabel>("Label").padding = new RectOffset(100, 0, 0, 0);
    group.AddCheckbox("My Checkbox 2", true, (b) => { });
}
```

![v2](https://user-images.githubusercontent.com/1386719/55296550-09f9fa00-5413-11e9-92eb-6de4f2a7304d.png)

Note that with this approach, you don't even need the group (so you can get rid of unwanted separator line that appears under groups)...

![v3](https://user-images.githubusercontent.com/1386719/55296564-25650500-5413-11e9-9ef0-ed8c4d102c98.png)

This should be ideal for the TMPE mod options and solve the overlapping options problem.

**Comment by aubergine10** [comment]
Hrm, looks like the code that adds the components to the UI would need to be passed info about the need to indent. No idea how to do that, sorry :/

**Comment by aubergine10** [comment]
Bug fix should be in next release thanks to @krzychu124  :)

**Comment by aubergine10** [comment]
Another report of this issue in Steam, only this time all options were missing (not just overlapping):

![xielBKi](https://user-images.githubusercontent.com/1386719/55650201-fd8feb80-57dc-11e9-8528-0210151036ef.jpg)

I've asked for log file just in case there are some other issues at play, but suspect that the bugfix by @krzychu124  will also fix the 'no options' issue too.

**Comment by aubergine10** [comment]
I'm going to reopen and repin this - until the fix is pushed to steam workshop it's still affecting users so we should keep this issue easy to find.

**Comment by aubergine10** [comment]
Fixed in 10.19 release
